crates/h264/src/boxes.rs

```diff
@@ -0,0 +1,119 @@
+//! ISO base media file format boxes for AVC/h264.
+//!
+//! ISO/IEC 14496-15 - 5.4
+
+use isobmff::boxes::{MPEG4ExtensionDescriptorsBox, VisualSampleEntry};
+use isobmff::{IsoBox, UnknownBox};
+
+use crate::AVCDecoderConfigurationRecord;
+
+/// AVC configuration box
+///
+/// ISO/IEC 14496-15 - 5.4.2
+#[derive(IsoBox, Debug, PartialEq, Eq)]
+#[iso_box(box_type = b"avcC")]
+pub struct AVCConfigurationBox<'a> {
+    /// The AVC decoder configuration record.
+    pub avc_config: AVCDecoderConfigurationRecord<'a>,
+}
+
+impl<'a> AVCConfigurationBox<'a> {
+    /// Creates a new AVC configuration box.
+    pub fn new(avc_config: AVCDecoderConfigurationRecord<'a>) -> Self {
+        Self { avc_config }
+    }
+}
+
+// This doesn't make sense to me, the following 4 box types (avc1 - avc4) are all the same
+
+/// AVC sample entry
+///
+/// ISO/IEC 14496-15 - 5.4.2
+#[derive(IsoBox, Debug, PartialEq, Eq)]
+#[iso_box(box_type = b"avc1")]
+pub struct AVCSampleEntry1<'a> {
+    /// The visual sample entry fields that this box inherits.
+    pub visual_sample_entry: VisualSampleEntry,
+    /// The AVC configuration box contained in this box.
+    #[iso_box(nested_box)]
+    pub config: AVCConfigurationBox<'a>,
+    /// The optional MPEG-4 extension descriptors box contained in this box.
+    #[iso_box(nested_box(collect))]
```

Comment:
not a fan, ideally we should just be able to say this is a nested_box without adding the collect part, likewise we should be able to do this for vecs without collect.